crates/sui-core/src/authority.rs

```diff
@@ -807,14 +807,26 @@ impl AuthorityState {
 
         // If commit_certificate returns an error, tx_guard will be dropped and the certificate
         // will be persisted in the log for later recovery.
-        self.commit_certificate(
-            inner_temporary_store,
-            certificate,
-            &signed_effects,
-            bypass_validator_halt,
-        )
-        .await
-        .tap_err(|e| error!(?digest, "commit_certificate failed: {}", e))?;
+        if let Err(err) = self
+            .commit_certificate(
+                inner_temporary_store,
+                certificate,
+                &signed_effects,
+                bypass_validator_halt,
+            )
+            .await
+        {
+            if matches!(err, SuiError::ValidatorHaltedAtEpochEnd) {
+                debug!(
+                    ?digest,
+                    "validator halted and this cert will never be committed"
+                );
+                tx_guard.release();
```

Comment:
this is fine, but as an alternate suggestion - if we pass the notifier ticket into commit_certificate, then we don't need the bypass_validator_halt argument. This will hoist the validator halted error check up into this function, and then commit_certificate can never fail to due to halt.

not a big deal either way but it seems like that might be cleaner.